pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -357,6 +372,44 @@ func getIPAddressStrings(ctx context.Context, c *config, addrName string) ([]str
 func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec, error) {
 	return spec, nil
 }
+func getStorageProfile(config *config, providerCfg *providerconfigtypes.Config) (*compute.StorageProfile, error) {
+	osRef, err := getOSImageReference(config.ImageID, providerCfg.OperatingSystem)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get OSImageReference: %v", err)
+	}
+	// initial default storage prfile, this will use the VMSize default storage profile
+	sp := &compute.StorageProfile{
+		ImageReference: osRef,
+	}
+	if config.OSDiskSize != "" && config.OSDiskSize != "0" {
+		osDiskSize, err := strconv.Atoi(config.OSDiskSize)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert OSDeskSize to int: %v", err)
+		}
+		osDiskSize32 := int32(osDiskSize)
+		sp.OsDisk = &compute.OSDisk{
+			DiskSizeGB:   &osDiskSize32,
```

Comment:
Nit: maybe using package `k8s.io/utils/pointer` once `config.OSDiskSize` is changed to int32 :) 
```go
 pointer.Int32Ptr(config.OSDiskSize),
```